pdp/policy.go

```diff
@@ -243,6 +247,52 @@ func (p *Policy) delChild(ID string) (*Policy, error) {
 	return p.updatedCopy(rules, algorithm), nil
 }
 
+// MarshalWithDepth implements StorageMarshal
+func (p Policy) MarshalWithDepth(out io.Writer, depth int) error {
+	if depth < 0 {
+		return newMarshalInvalidDepthError(depth)
+	}
+	pjson, err := json.Marshal(storageNodeFmt{
```

Comment:
It's better to make a function here like marshalHeader.
```go
func marshalHeader(v interface{}, w io.Writer) err {
	b, err := json.Marshal(v)
	if err != nil {
		return err
	}

	if len(b) < 1 || b[len(b)-1] != '}' {
		return newInvalidHeaderError()
	}

	_, err = w.Write(b[:len(b)-1])
	return err
}
```
And call it here and below.